src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -614,18 +615,13 @@ private AuthenticationResult processTokenResponse(HttpWebResponse webResponse) {
 
         // Set correlationId in the result
         if (correlationIdInHeader != null && !correlationIdInHeader.isEmpty()) {
-            try {
-                UUID correlation = UUID.fromString(correlationIdInHeader);
-                if (!correlation.equals(mRequest.getCorrelationId())) {
-                    Logger.w(TAG, "CorrelationId is not matching", "",
-                            ADALError.CORRELATION_ID_NOT_MATCHING_REQUEST_RESPONSE);
-                }
-
-                Logger.v(TAG, "Response correlationId:" + correlationIdInHeader);
-            } catch (Exception ex) {
-                Logger.e(TAG, "Wrong format of the correlation ID:" + correlationIdInHeader, "",
-                        ADALError.CORRELATION_ID_FORMAT, ex);
+            UUID correlation = UUID.fromString(correlationIdInHeader);
```

Comment:
UUID.fromString can throw IllegalArgumentException.  There's no reason that we should be attempting to convert the correlation id from the server since there's no reason that it has to be a string.

That's no a bug that you introduced, but removing the catch all exceptions clause will allow a new, potentially app crashing failure mode that didn't exist anymore.  You change is correct.  But now we should deal with the failure mode that shouldn't have been there in the first place.